### DISCUSSION TITLE
Author link issue not picking up on customized author-profile.html includes

### DISCUSSION THREAD

**Discussion by piyushpaliwal** [mmistakes/minimal-mistakes#2622] [Q&A] [answered]
## Environment

- Minimal Mistakes version: 4.19.3
- Ruby gem or remote theme version: 4.19.3
- Jekyll version: 3.8.6
- Git repository URL: https://github.com/piyushpaliwal/mmeverythingcrm
- GitHub Pages hosted (if yes provide URL to site):
- Operating system:

## Expected behavior

Author links should not show underscores as per the fix #2472. I have made modifications to the author profile page and looks like this particular fix is just not picking up during the build. I am pretty sure that it is a silly mistake that I may be doing but just can't seem to figure it out. I have tried Netlify clear up the cache and deploy new build but still the same.

This is what the code for the fix looks like on my repo 
[_includes/author-profile.html](https://github.com/piyushpaliwal/mmeverythingcrm/blob/02b1d17cf3a81addb45d0a402afa78c3d4666a8b/_includes/author-profile.html#L57) and below is the output hover state

![image](https://user-images.githubusercontent.com/635693/88472402-fb9cef00-cf2f-11ea-9f14-4f787017233b.png)

Any help will be greatly appreciated.

**Comment by mmistakes** [comment]
Check #2472 again, there was also a change made to `_sass/minimal-mistakes/_sidebar.scss`. That controls the styling which is why you're still seeing the underline on the space between the icon and link label.

**Comment by piyushpaliwal** [comment]
> Check #2472 again, there was also a change made to `_sass/minimal-mistakes/_sidebar.scss`. That controls the styling which is why you're still seeing the underline on the space between the icon and link label.

@mmistakes I understood that a css change was also done, but since I'm using the gem version and have updated to 4.19.3, any specific reason why this scss change is not getting picked up automatically?
I do understand the change that I have to manually do in the `author-profile.html` because of the customization.

**Comment by mmistakes** [comment]
Looking at your author-profile.html local file, [this line is your problem](https://github.com/piyushpaliwal/mmeverythingcrm/blob/master/_includes/author-profile.html#L47). It doesn't match what's current on `master` in the MM gem. You have a space between the icon and `link.label`, which is why the underline is showing up.

Line 47 should be this if you want to match the theme fix:

```
<li><a href="{{ link.url }}" rel="nofollow noopener noreferrer"><i class="{{ link.icon | default: 'fas fa-link' }}" aria-hidden="true"></i><span class="label">{{ link.label }}</span></a></li>
```

**Comment by piyushpaliwal** [comment]
Ok,
I understand but this extra space is not there for rest of the links, for
instance lets take Github link (
https://github.com/piyushpaliwal/mmeverythingcrm/blob/762d5ae189fab18b82af8546288dfab705bf22ce/_includes/author-profile.html#L140
)
There is no extra space here and still it doesn't respect the fix.

Regards,
Piyush

**Comment by mmistakes** [comment, marked as answer]
Don't wrap the lines. That is introducing the space. You may have to turn off autoformatting in your editor as:

```html
<li>
  <a href="https://github.com/{{ author.github }}" itemprop="sameAs" rel="nofollow noopener noreferrer">
    <i class="fab fa-fw fa-github" aria-hidden="true"></i><span class="label">GitHub</span>
  </a>
</li>
```

Isn't the same as:

```html
<li><a href="{{ link.url }}" rel="nofollow noopener noreferrer"><i class="{{ link.icon | default: 'fas fa-link' }}" aria-hidden="true"></i><span class="label">{{ link.label }}</span></a></li>
```

Your indents are being treated as white space. Hence the extra space.

**Comment by piyushpaliwal** [comment]
> Looking at your author-profile.html local file, [this line is your problem](https://github.com/piyushpaliwal/mmeverythingcrm/blob/master/_includes/author-profile.html#L47). It doesn't match what's current on `master` in the MM gem. You have a space between the icon and `link.label`, which is why the underline is showing up.
> 
> Line 47 should be this if you want to match the theme fix:
> 
> ```
> <li><a href="{{ link.url }}" rel="nofollow noopener noreferrer"><i class="{{ link.icon | default: 'fas fa-link' }}" aria-hidden="true"></i><span class="label">{{ link.label }}</span></a></li>
> ```

You are absolutely right, I knew it was a silly thing that I was missing out on; completely ignored the links loop. Apologies for the inconvenience, and thanks for the prompt responses. Really appreciate it.
